### DISCUSSION TITLE
Neo-Tree showing all files as hidden files

### DISCUSSION THREAD

**Discussion by ashermkn1** [nvim-neo-tree/neo-tree.nvim#808] [Q&A] [answered]
I am using LazyVim right out of the box with no custom config or anything, and when I open neo-tree with <leader>-e, it says that all my files are hidden, here is a screenshot of it happening in the ~/.config/nvim directory but it happens in any other directory I try. Is this a bug with what neo-tree thinks are hidden files or am I just being dumb?
<img width="284" alt="image" src="https://user-images.githubusercontent.com/28579754/224866029-f39c7fbd-2578-405f-99f6-b7ad8256920a.png">
and here it is after toggling the hidden files with "H"
<img width="286" alt="image" src="https://user-images.githubusercontent.com/28579754/224866176-6913cae2-616b-430c-b3f0-0b1c9f6c081b.png">

I asked the LazyVim community for help and they sent me here :)

**Comment by miversen33** [comment]
I suspect that Neo-tree believes the directory your in is hidden and thus it marks _everything_ under it as hidden. 

The standard in linux is that if something starts with `.`, its "hidden" and is thus treated as such. That said, I would think that if you opened that directory (or were already in it, etc), it _shouldn't_ mark everything under it as hidden.

That is my guess though. I don't use LazyNvim so I am not familiar with how to configure it. Thus I am going to assume you already know how to do this.

For the `neo-tree` part of that configuration, consider adding the following
```lua
filesystem = {
    filtered_items = {
        visible = true,
        hide_dotfiles = false,
        hide_hidden = false
    }
}
```

**Reply by miversen33** [reply]
For an example using lazy.nvim (the package manager that I assume LazyNvim uses), take a peek at [my dotfiles](https://github.com/miversen33/miversen-dotfiles/blob/475232790583dedbae1c207d1b91abf1c4e4df7d/editors/nvim/lua/plugins/init.lua#L961-L987)

```lua
        {
            "nvim-neo-tree/neo-tree.nvim", -- File Explorer
            branch = "v2.x",
            dependencies = {
                "nvim-lua/plenary.nvim",
                "kyazdani42/nvim-web-devicons", -- not strictly required, but recommended
                "MunifTanjim/nui.nvim",
            },
            dev = false,
            config = {
                popup_border_style = 'rounded',
                sources = {
                    "filesystem",
                },
                filesystem = {
                    filtered_items = {
                        visible = true,
                        hide_gitignored = false,
                        hide_hidden = false,
                        hide_dotfiles = false,
                    },
                    follow_current_file = true,
                },
            },
        },
```

**Reply by ashermkn1** [reply]
Huh, yeah I guess it's because the parent directory is hidden, although I was following [this](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwiGweylqdz9AhWAD1kFHQiPCRAQwqsBegQICRAF&url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DN93cTbtLCIM&usg=AOvVaw2OlvvQrYmN8LbIjEl-zQCM) tutorial and when he opened nvim in the `~/.config/nvim` directory (with LazyVim right out of the box as I have), only actual hidden files (`.gitignore` and `.neoconf.json`) were shown as hidden, which is weird.

**Reply by ashermkn1** [reply]
Although I just tested it in my `~` directory and it is marking every file as hidden
<img width="283" alt="image" src="https://user-images.githubusercontent.com/28579754/225138540-4ba9d169-8cea-474c-b0d6-e39bd64c6119.png">
So it can't be just the parent directory being hidden

**Comment by cseickel** [comment, marked as answer]
From the screenshot it looks like they are being hidden due to them being `gitgnore`d. Do you have a `gitignore` file that would cause this?

**Reply by ashermkn1** [reply]
Oh god I had a random `.gitignore` file in my `~` directory that had `*` as one of the lines. smh thank you so much lol past me is very stupid

**Reply by EthanNguyenPM** [reply]
Thank god, you save my time and life

**Reply by foreverLoveWisdom** [reply]
> Oh god I had a random `.gitignore` file in my `~` directory that had `*` as one of the lines. smh thank you so much lol past me is very stupid

Yes, it happens to me. Make sure to check if the folder/file is included in .gitignore or not 😅
